storage/src/main/java/org/apache/kafka/server/log/internals/FetchIsolation.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.server.log.internals;
+
+import org.apache.kafka.common.IsolationLevel;
+import org.apache.kafka.common.requests.FetchRequest;
+import org.apache.kafka.common.utils.FetchRequestUtils;
+
+public interface FetchIsolation {
```

Comment:
This should be an enum.

Reply:
A Java enum is the right way to represent `FetchIsolation`. In Scala, it's often emulated with a sealed trait and case objects. I don't understand the `javap` relevance here.

Reply:
And it's better if it's compiled to a class generally speaking.

Reply:
Ok, thanks. Should be fine now.